### PR TITLE
[Docs][ClientBuilder] Add `ray.client()` and `ray.ClientBuilder` to Experimental API docs

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
* Adds Docs about how to use `ray.client(...).connect()`!

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(
Docs are viewable here:
https://ray--16058.org.readthedocs.build/en/16058/package-ref.html#clientbuilder-api

### CODE REVIEW

File: python/ray/client_builder.py [hunk 1]
```diff
@@ -26,7 +26,8 @@ class ClientInfo:
 
 class ClientBuilder:
     """
-    Builder for a Ray Client connection.
+    Builder for a Ray Client connection. This class can be subclassed
+    by other builder classes.
```
Comment (DmitriGekhtman) [inline python/ray/client_builder.py:30]: Could help to explain how/why it should be subclassed.
For example could add "to modify connection behavior" at the end of the second sentence.

Comment (ijrsvt) [reply python/ray/client_builder.py:30]: Great idea!

**Comment by AmeerHajAli** [comment]
is this under experimental?

**Review by AmeerHajAli — APPROVED** [review]
a couple minor comments.

File: python/ray/client_builder.py [hunk 2]
```diff
@@ -36,17 +38,31 @@ def __init__(self, address: Optional[str]) -> None:
...
+        Args:
+            env (Dict[st, Any]): A runtime environment to use for this
+            connection. See ``runtime_env.py`` for what values are
+            accepted in this dict.
         """
         self._job_config.set_runtime_env(env)
         return self
 
     def namespace(self, namespace: str) -> "ClientBuilder":
+        """
```
[the remark this answers is not in the document]
Comment (ijrsvt) [reply python/ray/client_builder.py:50]: Namespaces have their own docs here: https://ray--16058.org.readthedocs.build/en/16058/namespaces.html

**Comment by ijrsvt** [comment]
> is this under experimental?

Yep!

**Review by DmitriGekhtman — APPROVED** [review]

**Comment by ijrsvt** [comment]
Failures are unrelated--this is a docs (/comments) only PR.

**Pull request merged**